### ISSUE TITLE
Script not running fast enough

### ISSUE THREAD

**Issue by martixy** [Tampermonkey/tampermonkey#1777]
Consider a site that does bad things to global objects. For example erasing the console. I might want to restore that functionality and use it. But this means injecting the script early, to preempt whatever fuckery the website is perpetrating.

However currently `@run-at` is almost entirely useless.
I think the reason is cache + async injection.

Now, I don't know exactly how the chrome HTML parser works.
But I know script tags are blocking, SSDs are fast, RAM is faster and the javascript event loop works in mysterious ways.

_Aren't race conditions fun?_

See the script below. Make sure you're loading from cache.

TL;DR: Maybe don't inject the script with an async function...? At least with `@run-at document-start`.

## Specifications

- Chromium: 113.0.5672.93 (Official Build) (64-bit)
- TM: v4.19.6183
- OS: Win10

## Script
```js
// ==UserScript==
// @name Test
// @namespace http://mywebsite.com
// @Version 1.0
// @description try to take over the world!
// @author Me
// @match https://www.linuxquestions.org/*
// @grant none
// @run-at document-start
// ==/UserScript==

console.log('waffles')
```
